composer/2022_airflow_summit/data_analytics_process_expansion.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import sys
+
+from py4j.protocol import Py4JJavaError
+from pyspark.sql import SparkSession
+from pyspark.sql.functions import avg, lit, sum, udf
+from pyspark.sql.types import FloatType, IntegerType, StructType
+
+# BQ_DESTINATION_DATASET_NAME = "expansion_project"
+# BQ_DESTINATION_TABLE_NAME = "ghcnd_stations_joined"
+# BQ_NORMALIZED_TABLE_NAME = "ghcnd_stations_normalized"
+# BQ_PRCP_MEAN_TABLE_NAME = "ghcnd_stations_prcp_mean"
+# BQ_SNOW_MEAN_TABLE_NAME = "ghcnd_stations_snow_mean"
+# BQ_PHX_PRCP_TABLE_NAME = "phx_annual_prcp"
+# BQ_PHX_SNOW_TABLE_NAME = "phx_annual_snow"
```

Comment:
Is this necessary to leave in for the purposes of the sample?

Reply:
I left it on purpose so that the PySpark program can run independently without having to run together with the DAG. You can check out the `If you just want to run the PySpark code:` section in my description :)